Source/Pawnmorphs/Esoteria/Hediffs/Composable/MutSpreadOrder.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Pawnmorph.Hediffs.Utility;
+using Pawnmorph.Utilities;
+using Pawnmorph.Utilities.Collections;
+using Verse;
+
+namespace Pawnmorph.Hediffs.Composable
+{
+    /// <summary>
+    /// A class that determines the order in which mutations spread through a person
+    /// </summary>
+    public abstract class MutSpreadOrder
```

Comment:
why a EquivalentTo method but not implement IEquatable<T>?

Reply:
I went back and forth on whether or not to use the equality methods, and decided against it since that would be a lot of overloading for what is honestly a mostly very specific use-case.  The only current use for EquivalentTo is to check whether the spread order cache should be purged on a stage changes.

Honestly, I'm still not convinced that spread order should be per-stage anyway.  I left it as-is now because it's harder to add stuff to the hediff itself, but I can't actually think of a use case for a hediff that changes spread orders midway through.

Reply:
this all makes sense, And I don't see a problem with not implementing IEquatable, more a curiosity